lib/BuildSystem/BuildSystem.cpp

```diff
@@ -3090,15 +3092,30 @@ class MkdirCommand : public ExternalCommand {
 };
 
 class MkdirTool : public Tool {
+  bool excludeFromOwnershipAnalysis = false;
+
 public:
   using Tool::Tool;
 
   virtual bool configureAttribute(const ConfigureContext& ctx, StringRef name,
                                   StringRef value) override {
-    // No supported attributes.
-    ctx.error("unexpected attribute: '" + name + "'");
-    return false;
+    if (name == "exclude-from-ownership-analysis") {
```

Comment:
This snippet of code that parses `exclude-from-ownership-analysis` from the manifest is repeated in SymlinkTool and other places. That's not ideal. Let's factor it out into a helper method